src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -29,6 +31,81 @@ constexpr auto DELAYED_EVENTS_QUEUE_PATH = "queue/vd/delayed";
 constexpr auto SLEEP_RETRY_THREADS_SEC = 60;
 
 int SOCKET_WAIT = 0;
+constexpr auto COMPRESSED_DB_PATH {"vd_1.0.0_vd_4.8.0.tar.xz"};
+constexpr auto DECOMPRESSED_DB_PATH {"vd_1.0.0_vd_4.8.0.tar"};
+constexpr auto VD_STATE_QUEUE_PATH = "queue/vd/state_track";
+constexpr auto VD_KEYSTORE_PATH = "queue/keystore";
+constexpr auto VD_DATABASE_PATH {"queue/vd"};
+constexpr auto VD_DATABASE_VERSION_KEY {"installed_content"};
+
+bool VulnerabilityScannerFacade::decompressDatabase(const std::string& databaseVersion)
+{
+    bool ret = false;
+
+    // Check database version and presence. It will attempt to decompress the database
+    // if the version doesn't not match, the state_track does not have information
+    // or there's no VD DB.
+    if ((databaseVersion.compare(__ossec_version) != 0 || databaseVersion.empty() ||
+         !std::filesystem::exists(VD_DATABASE_PATH)) &&
+        std::filesystem::exists(TMP_DIR))
+    {
+        // Search in path for a compressed database file.
+        std::regex regexPattern(VD_COMPRESSED_FILE_PATTERN);
+        std::string compressedFilePath;
+        size_t compressedFilesNumber = 0;
+        for (const auto& item : std::filesystem::directory_iterator(TMP_DIR))
+        {
+            if (std::regex_match(item.path().filename().string(), regexPattern))
+            {
+                logDebug2(WM_VULNSCAN_LOGTAG, "Found compressed database file: %s", item.path().c_str());
+                compressedFilePath = item.path().string();
+                break;
+            }
+        }
+
+        // Check for XZ downloaded file .
+        if (compressedFilePath.empty())
+        {
+            logWarn(WM_VULNSCAN_LOGTAG, "Missing database compressed file. Check DOWNLOAD_CONTENT option.");
+            return ret;
+        }
+
+        logInfo(WM_VULNSCAN_LOGTAG, "Starting database file decompression.");
+        logDebug2(WM_VULNSCAN_LOGTAG, "Starting XZ file decompression.");
+
+        std::string decompressedFilePath = compressedFilePath;
+        Utils::replaceFirst(decompressedFilePath, ".xz", "");
+
+        // Decompress XF file format.
+        Utils::XzHelper(std::filesystem::path(compressedFilePath), std::filesystem::path(decompressedFilePath))
+            .decompress();
+
+        // Extract queue/vd and queue/vd_updater
+        std::vector<std::string> extractOnly;
+        extractOnly.emplace_back(VD_DATABASE_PATH);
+
+        // Decompress also keystore if missing.
+        if (!std::filesystem::exists(VD_KEYSTORE_PATH))
+        {
+            extractOnly.emplace_back(VD_KEYSTORE_PATH);
+        }
+
+        logDebug2(WM_VULNSCAN_LOGTAG, "Starting TAR file decompression.");
+
+        // Decompress TAR file format.
+        Utils::ArchiveHelper::decompress(decompressedFilePath, "", extractOnly);
```

Comment:
We should clean (only the feed dbs) what we had previously, imagine the case of decompressing one database on top of another, this can cause corruption of the filesystem desired by rocksdb.

Reply:
If the feed dbs exists (vd actually) the decompression will not occur. It will decompress in case there's no vd information, the state_track does not have info or the Wazuh version does not match. That's why I didn't clean up that database before decompressing it. I'll refactor that to fix that behavior.

Reply:
Removing feed database if exists before decompression, the file will be decompressed if the version does not match or the state db has nothing. Everything else in vd if exist won't be touched.